pkgs/development/python-modules/glean-sdk/default.nix

```diff
@@ -0,0 +1,49 @@
+{ lib
+, pkgs
+, python3Packages
+, rustPlatform
+, setuptools-rust
+, fetchpatch
+}:
+
+python3Packages.buildPythonPackage rec {
+  pname = "glean-sdk";
+  version = "42.2.0";
+
+  src = python3Packages.fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-X2p6KQnEB6ZHdCHGFVEoEMiI+0R2vfGqel+jFKTcx74=";
+  };
+
+  patches = [
+    # Fix the environment for spawned process
+    # https://github.com/mozilla/glean/pull/1542
+    ./pr-1542.patch
```

Comment:
Can we fetch the patch using fetchpatch instead?

Reply:
That's what I did originally, but then changed course for the following reasons:
  1. patch is in a user fork. There is no guarantee that this branch is going to be alive tomorrow. It's quite possible that the branch will be deleted, and github will eventually return "not found" for the revision.
  2. whole patch from that PR is not applied correctly, it has conflicts.

So I think it's totally fine to just include this small patch, it will be removed on the next update.

Reply:
> There is no guarantee that this branch is going to be alive tomorrow.

That is not how github works. The patch will pretty much exist forever. There is no guarantee that this branch is going to be alive tomorrow.

> whole patch from that PR is not applied correctly, it has conflicts.

fetchpatch can exclude files if the issue is only in one file.

but it is also fine to vendor such a small patch. Anyhow please give it a proper name like the comment above instead pr-XXXX.